app/topics/show-topic/show-topic.component.ts

```diff
@@ -17,6 +17,8 @@ export class ShowTopicComponent implements OnInit, OnDestroy {
   title = '';
   userCanDelete: boolean = false;
   userCanEditDetails: boolean = false;
+  statusChange: boolean = false;
```

Comment:
can you think of a better name for that variable? What does it do?

When I get it correctly, you check with this flag if the user pressed the button to change the status? If so, this variable is obsolete.